web/react/components/rename_channel_modal.jsx

```diff
@@ -75,6 +75,9 @@ module.exports = React.createClass({
             }.bind(this)
         );
     },
+    handleClose: function() {
+        this.setState({display_name: "", channel_name: ""})
```

Comment:
This also needs to clear the error state. Repro case 1 from the ticket still is an issue.